### PR TITLE
Split the documentation pages into User guide and reference

### PR DESCRIPTION
**Pull request by ppalaga**
This still needs some polishing. Before I invest too much time into it, I'd like to hear your opinion @shumonsharif and @famod whether the restructuring of the docs into 

1. User guide containing some recipes for solving typical CXF related tasks and 
2. Reference - a list of our extensions and their config options

makes sense to you?

You can view the generated site locally by running

```
$ cd docs 
$ mvn test
$ google-chrome target/site/quarkus-cxf/dev/index.html
```

### CODE REVIEW

**Comment by ppalaga** [comment]
@shumonsharif @famod in case you have Windows or Mac, it would be great to know whether `cd docs && mvn clean test` works for you? 

I would not necessarily want to fix it. I'd say setting some Junit assumes for skipping on Max/Windows would fine. WDYT?

**Comment by ppalaga** [comment]
BTW, the CI job is now publishing the generated Antora site as a zip file:
image

**Comment by shumonsharif** [comment]
Hi @ppalaga Sorry for the late response, it's been a hectic couple days at the day job.  This is looking great, and I'm all for it; I think it's a simpler layout than we previously had - and therefore more consumable.

On my Mac, `cd docs && mvn clean test` works fine, and the pages are generated without issue.

One note worth considering - we did have an open ticket for a quarkus-cxf quickstart (https://github.com/quarkiverse/quarkus-cxf/issues/399) - I personally learn much faster about any given framework or extension by looking at example code, not as much through docs.  I'm wondering if it's worth revisiting the quickstart issue while also updating the docs.

**Comment by ppalaga** [comment]
> On my Mac, cd docs && mvn clean test works fine, and the pages are generated without issue.

Great to hear that! 
Now, let's wait for the first person who will complain that it does not work on Windows ;-)

> we did have an open ticket for a quarkus-cxf quickstart (#399) - I personally learn much faster about any given framework or extension by looking at example code, not as much through docs. I'm wondering if it's worth revisiting the quickstart issue while also updating the docs.

I am all for improving the learning experience by all means. I got a bit lost by reading #399. Could you please sum up what is the latest objective there?

**Comment by ppalaga** [comment]
The last iteration ending with fa04a19 is still not perfect from my PoV, but already good enough to be merged. 

It would be great if somebody could review the content.

**Comment by shumonsharif** [comment]
> I am all for improving the learning experience by all means. I got a bit lost by reading #399. Could you please sum up what is the latest objective there?

Hey @ppalaga The basic idea was to provide sample project(s) that would supplement the documentation.  So, if for example, a user wanted to see a WS-Trust implementation and how it would work with `quarkus-cxf`, there would be a pre-canned project they can refer to without going through all the docs or having to browse through the integration tests.

As far as summing up #399, the initial intent was to build out a set of [codestarts](https://quarkus.io/blog/tag/codestarts/) (Andy's blog write-ups sums up codestarts nicely).  That would have been really nice if we could integrate, but after all the back-and-forth, we agreed a set of `quickstart`s may be more appropriate - like Quarkus provides [here](https://github.com/quarkusio/quarkus-quickstarts).

**Review by shumonsharif — COMMENTED** [review]
Hi @ppalaga Just a few comments, looks great overall!

**Comment by famod** [comment]
Got that yesterday on Winodws:
```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.031 s <<< FAILURE! - in io.quarkiverse.cxf.doc.it.AntoraTest
[ERROR] antora  Time elapsed: 0.016 s  <<< ERROR!
java.lang.UnsupportedOperationException: View 'unix' not available
        at java.base/sun.nio.fs.AbstractFileSystemProvider.readAttributes(AbstractFileSystemProvider.java:93)
        at java.base/java.nio.file.Files.readAttributes(Files.java:2084)
        at java.base/java.nio.file.Files.getAttribute(Files.java:1983)
        at io.quarkiverse.cxf.doc.it.AntoraTest.antora(AntoraTest.java:47)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
```
But I don't mind as I'm working on Ubuntu in WSL2 (private PC) or Ubuntu natively (work laptop) anyway.

**Comment by shumonsharif** [comment]
> But I don't mind as I'm working in Ubuntu via WSL2 (private PC) or Ubuntu natively (work laptop) anyway.

My only concern with the Windows build would be not so much for the core group here, but more to allow public contributors to be able to participate with ease.  Granted, we don't get too many such PRs ... :) I think we can skip the test on Windows as @ppalaga suggested, but still curious why the unix uid is needed to begin with?

**Comment by shumonsharif** [comment]
So, I found the answer to my question at link below, seems like it's to address volume mounts and SELinux issues.
https://docs.antora.org/antora/latest/antora-container/#option-flags

I'm now wondering if it's possible to to simply catch all exceptions when trying to identify the uid, and use the uid value only if it exists.  I believe that should should allow the test / doc-generation to also work on Windows without issues.

File: docs/src/test/java/io/quarkiverse/cxf/doc/it/AntoraTest.java [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+
+        final Path targetDir = Paths.get("target");
+        if (!Files.isDirectory(targetDir)) {
+            try {
+                Files.createDirectories(targetDir);
+            } catch (IOException e) {
+                throw new RuntimeException("Could not create " + targetDir);
+            }
+        }
+        final int uid;
```
[the remark this answers is not in the document]
Comment (shumonsharif) [reply docs/src/test/java/io/quarkiverse/cxf/doc/it/AntoraTest.java:45]: Pls refer to https://github.com/quarkiverse/quarkus-cxf/pull/691#issuecomment-1399331089 and https://github.com/quarkiverse/quarkus-cxf/pull/691#issuecomment-1399497748 ... figured out why the uid is needed, but wondering if we can make this test work for Windows too.

[the remark this answers is not in the document]
Comment (famod) [reply docs/src/test/java/io/quarkiverse/cxf/doc/it/AntoraTest.java:45]: The test passes on my Windows box when commenting out everything related to `uid`.

[the remark this answers is not in the document]
Comment (shumonsharif) [reply docs/src/test/java/io/quarkiverse/cxf/doc/it/AntoraTest.java:45]: Hey @famod Thank you so much for testing!  Are the permissions on the `target` and its subfolders ok?  ie. hoping that they aren't created under some Windows account other than the host machine's user?  so `mvn clean` should theoretically work fine after the initial `mvn clean test`.

[the remark this answers is not in the document]
Comment (famod) [reply docs/src/test/java/io/quarkiverse/cxf/doc/it/AntoraTest.java:45]: Gotcha! Subsequent run is fine and permissions look ok to me.

**Comment by ppalaga** [comment]
> As far as summing up #399, the initial intent was to build out a set of [codestarts](https://quarkus.io/blog/tag/codestarts/) (Andy's blog write-ups sums up codestarts nicely). That would have been really nice if we could integrate, but after all the back-and-forth, we agreed a set of `quickstart`s may be more appropriate - like Quarkus provides [here](https://github.com/quarkusio/quarkus-quickstarts).

Thanks for explaining. Mainly the role of codestarts was unclear to me. 

As a maintainer of several current or past projects, where a separate quickstarts repo played a role, I must say I am not a big fan of them. It is a bit of overhead to keep them up to date with the main repo (CI) and a few more steps when releasing. I like it much more when integration tests are sold as quickstarts. Those are in the same source tree, they must work before and after each merged change, and it is only good (for both maintainers and end users) if they are well documented through READMEs and inline comments. Snippets from those can also be used in Antora docs, which I started doing with this PR.

Having said that I won't block you or anybody else from creating a separate Quickstarts repo. It would be nice if the additional release steps could be documented somewhere (a new release guide document maybe?)

File: docs/src/test/java/io/quarkiverse/cxf/doc/it/AntoraTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ppalaga) [reply docs/src/test/java/io/quarkiverse/cxf/doc/it/AntoraTest.java:45]: Thanks @famod for testing!

Yes, the uid stuff is mainly to be able to delete the files via `mvn clean`. 

Let me try to come up with some try or if to make it work also on Windows.

File: integration-tests/server/${maven.multiModuleProjectDirectory}/docs/modules/ROOT/examples/hello-world/HelloService.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package io.quarkiverse.cxf.it.server;
```
[the remark this answers is not in the document]
Comment (ppalaga) [reply integration-tests/server/${maven.multiModuleProjectDirectory}/docs/modules/ROOT/examples/hello-world/HelloService.java:1]: good catch, thanks! Indeed, the files should go to docs/modules/ROOT/examples. Let me fix it.

File: docs/modules/ROOT/pages/user-guide/first-soap-client.adoc [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+To make it testable easily, we'll wrap it in a REST service:
+
+.CxfClientResource.java
+[source,java]
+----
+include::example$calculator-client/CxfClientRestResource.java[]
+----
+
+<1> Let the CDI container inject in instance of the client. `@CXFClient("myCalculator")` is actually equivalent to `@Inject @CXFClient("myCalculator")`
```
[the remark this answers is not in the document]
Comment (ppalaga) [reply docs/modules/ROOT/pages/user-guide/first-soap-client.adoc:143]: Thanks, fixed.

**Comment by shumonsharif** [comment]
> As a maintainer of several current or past projects, where a separate quickstarts repo played a role, I must say I am not a big fan of them. It is a bit of overhead to keep them up to date with the main repo (CI) and a few more steps when releasing. I like it much more when integration tests are sold as quickstarts. Those are in the same source tree, they must work before and after each merged change, and it is only good (for both maintainers and end users) if they are well documented through READMEs and inline comments. Snippets from those can also be used in Antora docs, which I started doing with this PR.

Makes sense - I thought of some of these issues, but never having actually implemented any quickstarts, I wasn't sure on how the release process would work exactly ... definitely seems like more trouble than it's worth.  Thanks for your input on this!

**Pull request merged**